src/storage/IpfsStorage.ts

```diff
@@ -63,8 +65,23 @@ export class IpfsStorage implements IStorage {
     return `ipfs://${cid}/`;
   }
 
+  public async uploadBatchWithFileNames(
+    files: FileOrBufferWithNames[],
+    contractAddress?: string,
+  ): Promise<string> {
+    const cid = await this.uploadBatchWithCid(true, files, contractAddress);
+
+    return `ipfs://${cid}/`;
+  }
+
   private async uploadBatchWithCid(
-    files: Buffer[] | string[] | FileOrBuffer[] | File[],
+    withFileNames = false,
+    files:
```

Comment:
I don't like this approach with the `files` field taking on so many different types in the union.

Move on to another ticket for now and I can come in and clean this interface up.